File: sdk/php/tests/conformance.php
<?php
declare(strict_types=1);

require_once __DIR__ . '/../src/Webhook.php';

use Supaapps\Platform93\Webhook;

$fixture = json_decode((string) file_get_contents(__DIR__ . '/../../../conformance/webhook.json'), true, 32, JSON_THROW_ON_ERROR);
$event = Webhook::verify($fixture['raw_body'], $fixture['signature'], $fixture['secret'], 1_000_000_000);
if (($event['id'] ?? null) !== '01900000-0000-7000-8000-000000000001') {
    throw new RuntimeException('Shared webhook fixture failed.');
}
if (!Webhook::isPlatformEvent($event)) throw new RuntimeException('Platform event discrimination failed.');
Webhook::assertSupportedPlatformEvent($event);

File: sdk/python/webhooks/tests/conformance.py
import json
from pathlib import Path

from platform93_webhooks import assert_supported_platform_event, is_platform_event, verify_webhook


fixture = json.loads(
    (Path(__file__).parents[4] / "conformance" / "webhook.json").read_text()
)
event = verify_webhook(
    fixture["raw_body"].encode(),
    fixture["signature"],
    fixture["secret"],
    now=fixture["timestamp"],
)
assert event["id"] == "01900000-0000-7000-8000-000000000001"
assert is_platform_event(event)
assert_supported_platform_event(event)
